types/raygun/index.d.ts

```diff
@@ -0,0 +1,116 @@
+// Type definitions for raygun 0.10
+// Project: https://github.com/MindscapeHQ/raygun4node
+// Definitions by: Taylor Lodge <https://github.com/UberMouse>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+
+declare namespace raygun {
+    interface KeyValueObject  {
+        [key: string]: string | number | boolean | KeyValueObject;
+    }
+    interface StackFrame {
+        lineNumber: number;
+        className: string;
+        fileName: string;
+        methodName: string;
+        columnNumber?: number;
+    }
+    interface RaygunErrorObject {
+        message: string;
+        className: string;
+        stackTrace: StackFrame[];
+        innerError?: RaygunErrorObject;
+    }
+    interface RaygunRequest {
+        hostname?: string;
+        host?: string;
+        path?: string;
+        method?: string;
+        ip?: string;
+        queryString?: KeyValueObject;
+        headers?: KeyValueObject;
+        form?: KeyValueObject;
+    }
+    interface RaygunUser {
+        identifier: string;
+        email?: string;
+        fullName?: string;
+        firstName?: string;
+        uuid?: string;
+    }
+    interface RaygunPayload {
+        occurredOn: Date;
+        details: {
+            client: {
+                name: 'raygun-node';
+                version: string;
+            };
+            groupingKey?: string;
+            error: RaygunErrorObject;
+            environment: {
+                osVersion: string;
+                architecture: string;
+                totalPhysicalMemory: number;
+                availablePhysicalMemory: number;
+                utcOffset: number;
+                processorCount?: number;
+                cpu: {
+                    model: string;
+                    speed: number;
+                    times: {
+                        user: number;
+                        nice: number;
+                        sys: number;
+                        idle: number;
+                        irq: number;
+                    };
+                };
+            };
+            machineName: string;
+            userCustomData?: KeyValueObject;
+            tags: string[];
+            request?: RaygunRequest;
+            user?: RaygunUser | { identifier: string }
+            version?: string;
+        };
+    }
+
+    interface RaygunOfflineStorageProvider<TTransportItem = RaygunPayload, TStorageItem = string> {
+        init(options: any): RaygunOfflineStorageProvider;
```

Comment:
Nit, avoid `any`.

Reply:
Should I use `object` instead then? Have to cater for customers providing their own offline storage provider which could take anything as options

Reply:
Could make it generic eg. `RaygunOfflineStorageProvider<TTransportItem = RaygunPayload, TStorageItem = string, TOptions = void>`, although that might lead to having to bubble up the generic...

Reply:
`any` (or `unknown`) is fine if it really is anything